dist/s_docs

```diff
@@ -192,6 +186,23 @@ check_docs_data()
 
 build()
 {
+    # We require doxygen which may not be installed.
```

Comment:
Can we move this code section into a standalone `verify_doxygen_installed` function and run it just before `build $clean` on line 307?

Reply:
I can do it but why is it useful?

Reply:
I liked it for stylistic reasons - separation of concerns - but as discussed I'm happy to leave as is. I'll resolve this comment.